src/functionalities.js

```diff
@@ -0,0 +1,134 @@
+class ToDoList {
+  constructor() {
+    const form = document.querySelector('.add-task-form');
+    form.addEventListener('submit', (event) => {
+      event.preventDefault();
+      this.addTask();
+    });
+    this.allTasks = [];
+    this.storageName = 'to-do-list';
+    this.loadTasks();
+  }
+
+  // Add a new task to the Task list --> CORE REQUREMENT
+  addTask() {
+    const desc = document.querySelector('#task');
+    const task = {
+      description: desc.value,
+      completed: false,
+      index: this.allTasks.length,
+    };
```

Comment:
- You have done well so far, but there is just a little issue left, according to the project requirements, `By default, new tasks should have the property 'completed' set to 'false' and the property 'index' set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5)`. The indexing does not work as required in your project. Please fix this 🙏 

Your project:
- Please note that there are 2 tasks in the array but the index for the last task is set to 1 instead of 2.
<img width="958" alt="wtw4ta" src="https://github.com/M-Anwar-Hussaini/To-Do-List/assets/73607512/2e795806-9f20-4e59-aa1b-b60b68788f63">